### PR TITLE
[AEROGEAR-2733] Add unbound/bound services chart to mobile client

### PR DESCRIPTION
**Pull request by dimitraz**
**Description**
List all bound and unbound services in a doughnut graphic

**Verification Steps**
- Provision a mobile client and a mobile service
- Donut should show up in the mobile client row once the service is successfully provisioned
- Verify that the unbound and bound services numbers are correct
- Bind the mobile app to the mobile service
- Verify that the donut has been updated correctly

**Related Jira**
[AEROGEAR-2733](https://issues.jboss.org/browse/AEROGEAR-2733)

### CODE REVIEW

**Comment by dimitraz** [comment]
@JameelB

File: app/scripts/directives/overview/mobileClientRow.js [hunk 1]
```diff
@@ -67,6 +67,26 @@
     row.updateServicesInfo = function() {
       if (row.services) {
         row.servicesNotBoundCount = row.services.length - row.bindings.length;
+        var boundServices = 'Bound Services';
```
Comment (sedroche) [inline app/scripts/directives/overview/mobileClientRow.js]: Can we move the donut configuration/setup to `$onInit`? I think only `row.data` needs to be updated every change.

File: app/scripts/directives/overview/mobileClientRow.js [hunk 2]
```diff
@@ -67,6 +67,26 @@
...
+        row.config = {
+          'chartId': 'services-info',
+          'legend': {
+            'show': true,
+            'position': 'right'
+          },
+          colors: {}
+        };
+
+        row.config.colors[boundServices] = '#47b0d6';
```
Comment (sedroche) [inline app/scripts/directives/overview/mobileClientRow.js]: There is a palette available at `$.pfPaletteColors`. Can you check if that has suitable colours?

File: app/styles/_mobile-clients.less [hunk 3]
```diff
@@ -39,4 +39,8 @@
       }
     }
   }
+  .services-chart {
+    width: 200px;
```
Comment (sedroche) [inline app/styles/_mobile-clients.less]: css should be alphabetical

**Review by sedroche — APPROVED** [review]

File: app/scripts/directives/overview/mobileClientRow.js [hunk 4]
```diff
@@ -29,12 +29,27 @@
     var serviceBindingsVersion = APIService.getPreferredVersion('servicebindings');
     var isServiceInstanceReady = $filter('isServiceInstanceReady');
     var isMobileService = $filter('isMobileService');
+    var boundServices = 'Bound Services';
```
Comment (ghost) [inline app/scripts/directives/overview/mobileClientRow.js]: `const BOUND_SERVICES = 'Bound Services';` ?

File: app/scripts/directives/overview/mobileClientRow.js [hunk 5]
```diff
@@ -29,12 +29,27 @@
...
+    var unboundServices = 'Unbound Services';
```
Comment (ghost) [inline app/scripts/directives/overview/mobileClientRow.js]: `const UNBOUND_SERVICES = 'Bound Services';` ?

File: app/scripts/directives/overview/mobileClientRow.js [hunk 6]
```diff
@@ -29,12 +29,27 @@
...
+        'chartId': 'services-info',
+        'legend': {
+          'show': true,
+          'position': 'right'
+        },
+        colors: {}
+      };
+      row.config.colors[boundServices] = $.pfPaletteColors.blue;
+      row.config.colors[unboundServices] = $.pfPaletteColors.orange;
+      row.chartHeight = 80;
```
Comment (ghost) [inline app/scripts/directives/overview/mobileClientRow.js]:
```js
const CHAR_HEIGHT=80
row.charHeight=HEIGHT
```

**Review by vchepeli — CHANGES_REQUESTED** [review]

**Comment by vchepeli** [comment]
@dimitraz LGTM, small fixes

**Comment by dimitraz** [comment]
@sedroche info icon added

File: app/scripts/directives/overview/mobileClientRow.js (hunk 4 above)
Comment (dimitraz) [reply app/scripts/directives/overview/mobileClientRow.js]: They don't seem to use constants anywhere else in the codebase so for now we haven't added this

**Comment by dimitraz** [comment]
@sedroche @vchepeli rebased can you take another look

**Review by sedroche — APPROVED** [review]

**Review by vchepeli — APPROVED** [review]

**Pull request merged**